tools/pubsys/src/aws/ami/register/mk_amispec.rs

```diff
@@ -0,0 +1,833 @@
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp3::builder()
+                        .snapshot_id("snapshot!")
+                        .volume_size(10).unwrap()
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+            .build(),
+        hashmap! {
+            "friendly-volume-name".into() => BlockDeviceRenderContext {
+                device_name: "doesn't matter, only snap ID does".into(),
+                snapshot_id: "snapshot!".into(),
+                volume_size: 2
+            },
+            "unrelated-volume-name".into() => BlockDeviceRenderContext {
+                device_name: "doesn't matter, only snap ID does".into(),
+                snapshot_id: "different-snapshot!".into(),
+                volume_size: 20
+            },
+        },
+        AmiSpec::builder()
+            .name("test-bdm-bigger-than-snapshot")
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp3::builder()
+                        .snapshot_id("snapshot!")
+                        .volume_size(10).unwrap()
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+
+            .build()
+        ; "block device mapping larger than image"
+    )]
+    #[test_case(
+        AmiSpec::builder()
+            .name("test-bdm-snapshot-unrelated")
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp3::builder()
+                        .snapshot_id("some-random-snapshot")
+                        .volume_size(1).unwrap()
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+            .build(),
+        hashmap! {
+            "friendly-volume-name".into() => BlockDeviceRenderContext {
+                device_name: "doesn't matter, only snap ID does".into(),
+                snapshot_id: "snapshot!".into(),
+                volume_size: 2
+            },
+            "unrelated-volume-name".into() => BlockDeviceRenderContext {
+                device_name: "doesn't matter, only snap ID does".into(),
+                snapshot_id: "different-snapshot!".into(),
+                volume_size: 20
+            },
+        },
+        AmiSpec::builder()
+            .name("test-bdm-snapshot-unrelated")
+            .block_device_mapping(BlockDeviceMapping::builder()
+                .ebs(
+                    ebs::Gp3::builder()
+                        .snapshot_id("some-random-snapshot")
+                        .volume_size(1).unwrap()
+                        .build()
+                )
+                .build()
+                .with_device_name("/dev/xvda"))
+
+            .build()
+        ; "snapshots with no hints are ignored"
```

Comment:
What does "snapshots with no hints" mean? Are the two snapshots in the hash map ignored because neither of them has a matching snapshot ID? Otherwise, they both seem to have a volume size.